engine/rpcserver.go

```diff
@@ -4106,3 +4109,241 @@ func (s *RPCServer) CurrencyStateTradingPair(_ context.Context, r *gctrpc.Curren
+		}
+		for j := range pos[i].Orders {
+			var trades []*gctrpc.TradeHistory
+			for k := range pos[i].Orders[j].Trades {
+				trades = append(trades, &gctrpc.TradeHistory{
+					CreationTime: pos[i].Orders[j].Trades[k].Timestamp.Unix(),
+					Id:           pos[i].Orders[j].Trades[k].TID,
+					Price:        pos[i].Orders[j].Trades[k].Price,
+					Amount:       pos[i].Orders[j].Trades[k].Amount,
+					Exchange:     pos[i].Orders[j].Trades[k].Exchange,
+					AssetType:    pos[i].Asset.String(),
+					OrderSide:    pos[i].Orders[j].Trades[k].Side.String(),
+					Fee:          pos[i].Orders[j].Trades[k].Fee,
+					Total:        pos[i].Orders[j].Trades[k].Total,
+				})
+			}
+			details.Orders = append(details.Orders, &gctrpc.OrderDetails{
+				Exchange:      pos[i].Orders[j].Exchange,
+				Id:            pos[i].Orders[j].ID,
+				ClientOrderId: pos[i].Orders[j].ClientOrderID,
+				BaseCurrency:  pos[i].Orders[j].Pair.Base.String(),
+				QuoteCurrency: pos[i].Orders[j].Pair.Quote.String(),
+				AssetType:     pos[i].Orders[j].AssetType.String(),
+				OrderSide:     pos[i].Orders[j].Side.String(),
+				OrderType:     pos[i].Orders[j].Type.String(),
+				CreationTime:  pos[i].Orders[j].Date.Unix(),
+				UpdateTime:    pos[i].Orders[j].LastUpdated.Unix(),
+				Status:        pos[i].Orders[j].Status.String(),
+				Price:         pos[i].Orders[j].Price,
+				Amount:        pos[i].Orders[j].Amount,
+				Fee:           pos[i].Orders[j].Fee,
+				Cost:          pos[i].Orders[j].Cost,
+				Trades:        trades,
+			})
+		}
+		response.Positions = append(response.Positions, details)
+	}
+
+	if !totalUnrealisedPNL.IsZero() {
+		response.TotalUnrealisedPNL = totalUnrealisedPNL.String()
+	}
+	if !totalRealisedPNL.IsZero() {
+		response.TotalRealisedPNL = totalRealisedPNL.String()
+	}
+	if !totalUnrealisedPNL.IsZero() && !totalRealisedPNL.IsZero() {
+		response.TotalPNL = totalRealisedPNL.Add(totalUnrealisedPNL).String()
+	}
+	return response, nil
+}
+
+// GetCollateral returns the total collateral for an exchange's asset
+// as exchanges can scale collateral and represent it in a singular currency,
+// a user can opt to include a breakdown by currency
+func (s *RPCServer) GetCollateral(ctx context.Context, r *gctrpc.GetCollateralRequest) (*gctrpc.GetCollateralResponse, error) {
+	exch, err := s.GetExchangeByName(r.Exchange)
+	if err != nil {
+		return nil, err
+	}
+
+	a := asset.Item(r.Asset)
+	err = checkParams(r.Exchange, exch, a, currency.Pair{})
+	if err != nil {
+		return nil, err
+	}
+	if !a.IsFutures() {
+		return nil, fmt.Errorf("%s %w", a, order.ErrNotFuturesAsset)
+	}
+	ai, err := exch.FetchAccountInfo(ctx, a)
+	if err != nil {
+		return nil, err
+	}
+	var calculators []order.CollateralCalculator
+	var acc *account.SubAccount
+	if r.SubAccount != "" {
+		for i := range ai.Accounts {
+			if strings.EqualFold(r.SubAccount, ai.Accounts[i].ID) {
+				acc = &ai.Accounts[i]
+				break
+			}
+		}
+	} else if len(ai.Accounts) > 0 {
+		acc = &ai.Accounts[0]
```

Comment:
This is causing an account mismatch. Also out of scope issue with hold values when util exch.FetchAccountInfo(ctx, a) above as discussed in PM.